pkg/resourcecollector/persistentvolume.go

```diff
@@ -90,10 +92,43 @@ func (r *ResourceCollector) pvToBeCollected(
 func (r *ResourceCollector) preparePVResourceForCollection(
 	object runtime.Unstructured,
 ) error {
+	var pv v1.PersistentVolume
+	var currentSc string
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(object.UnstructuredContent(), &pv); err != nil {
+		return err
+	}
+	// Some time pv spec does not contains the storage class.
+	// In that case, we will get it from pvc spec.
+	if len(pv.Spec.StorageClassName) == 0 {
+		pvcName := pv.Spec.ClaimRef.Name
+		pvcNamespace := pv.Spec.ClaimRef.Namespace
+		pvc, err := r.coreOps.GetPersistentVolumeClaim(pvcName, pvcNamespace)
+		if err != nil {
+			return err
+		}
+		currentSc, err = utils.GetStorageClassNameForPVC(pvc)
+		if err != nil {
+			return err
+		}
+	} else {
+		currentSc = pv.Spec.StorageClassName
+	}
 	err := unstructured.SetNestedField(object.UnstructuredContent(), nil, "spec", "claimRef")
 	if err != nil {
 		return err
 	}
+	annotations, found, err := unstructured.NestedStringMap(object.UnstructuredContent(), "metadata", "annotations")
```

Comment:
Set the annotation only if you find a storage class name successfully. Check previous comment.

Reply:
Fixed